### ISSUE TITLE
optim-update-sequence-range-visiblity-once

### ISSUE THREAD

**Issue by aminere** [eternagame/EternaJS#556]
There is no need to traverse the whole sequence on every update and call `isNucleotidePartOfSequence` just to update the bases visibility. This can be done once in `Pose2D.showNucleotideRange`

**Comment by luxaritas** [comment]
Is this also used to determine whether a base is a "cut" base in multistrand puzzles?

**Comment by luxaritas** [comment]
Looking at the last commit looks like not, since it's behind the showNucleotideRange flag

**Comment by aminere** [comment]
Yes indeed, 

However, the base type doesn't seem to change throughout the app as I can't find any `type = RNABase.Cut` assignment. If you can confirm this, we can maybe get away with initializing the visibility, to take into account `RNABase.Cut`, and we still won't need to call it on every update.

**Comment by luxaritas** [comment]
As currently implemented, I think it will change if you change oligo or oligos - for example, in a multistate puzzle, state 1 may just have the main strand, but toggling to state 2 may add a cut and a second strand that includes 20 bases, state 3 has cut/10 bases/cut/20 bases, etc. I do remember having to fix a bug where cut bases were not properly hidden, but admittedly that was some time ago and before the addition of SecStruct and the updated RNABase

**Comment by luxaritas** [comment]
Yeah actually since this check only happens when show nucleotide range is false, I'm betting this is what that's for - and it's even incorrect for multistrand puzzles. I agree it most likely should not be done every frame (this may have been my fault due to improper understanding of dirty flags early on), but we do need to make sure it's handled properly

**Comment by luxaritas** [comment]
See 46ae2843eccea452429c2c7253e8141d4ea6a80f
